conans/client/generators/cmake_multi.py

```diff
@@ -50,7 +50,7 @@ def _content_type(self, build_type):
 
         # Per requirement variables
         for _, dep_cpp_info in self.deps_build_info.dependencies:
-            dep_name = dep_cpp_info.name
+            dep_name = dep_cpp_info.get_name("cmake")
```

Comment:
```suggestion
            dep_name = dep_cpp_info.get_name("cmake_multi")
```

Reply:
I start to provide a name by cmake generator, but I roll back my change after test it on my package. Indeed, cmake have many generators, but this is always CMake that consume generated file/code. And we have create this feature to have a package name diferente in function of build tools that use the generated file/code.
For example, freetype provide pkg-config config file named freetype2.pc, but CMake user are looking for a file named FindFreetype.cmake or Freetype-config.cmake, and conan package is named freetype. 
All pkg-config generator have to create a freetype2.pc, ok there are only one pkg-config ...
But all cmake generator have to create a FindFreetype.cmake or Freetype-config.cmake ...
So no need to hurt conan recipe writer to specify each cmake generator to use "Freetype" name. 
We should not provide a name by generator, but by generator family.
thought ?

Reply:
Hi @ledocc,
Our idea was to give the freedom to specify the name by each generator name, so you can have for example a conanfile.py like this:
```
        self.cpp_info.names["cmake"] = "My_cmake_customName"
        self.cpp_info.names["cmake_paths"] = "My_cmake_paths_customName"
        self.cpp_info.names["cmake_multi"] = "My_cmake_multi_paths_customName"
        self.cpp_info.names["cmake_find_package"] = "My_cmake_find_package_customName"
        self.cpp_info.names["cmake_find_package_multi"] = "My_cmake_find_package_multi_customName"
        self.cpp_info.names["pkg_config"] = "My_pkg_config_customName"
```

Reply:
Yes, but is it really usefull to have many name for each cmake generator ? This is always cmake that use code/files generated by cmake{_*} generators, and cmake work the same way when you use cmake, cmake_paths or cmake_find_package_multi generator. CMake and pkg-config require different name,  but cmake_paths and cmake_find_package need not.
Have you an use case where different cmake generator need different names ?

Reply:
We think that this is more flexible and that fits better with the current implementation of the generators, defining a `cpp_info.names` using the key `cmake` and affecting other generators will be confusing. 
I know what you mean, and I like the idea but I think that's something we need to consider for Conan 2.0 but not for this moment.

Reply:
ok